metastore/src/test/scala/com/gu/tableversions/metastore/MetastoreSpec.scala

```diff
@@ -154,34 +156,43 @@ trait MetastoreSpec {
     it should "return an error if trying to get the version of an unknown table" in {
       val scenario = for {
         metastore <- emptyMetastore
-        _ <- initHiveTable
+        _ <- initUnderlyingTable
 
         version <- metastore.currentVersion(TableName("unknown", "table"))
 
       } yield version
 
-      val ex = the[Exception] thrownBy scenario.unsafeRunSync()
+      val ex = the[Exception] thrownBy scenario.guarantee(tearDownUnderlyingTable).unsafeRunSync()
       ex.getMessage.toLowerCase should include regex "unknown.*not found"
     }
 
     it should "not allow updating the version of an unknown partition" in {
       val scenario = for {
         metastore <- emptyMetastore
-        _ <- initHiveTable
+        _ <- initUnderlyingTable
+        initialVersion <- metastore.currentVersion(table)
         version <- Version.generateVersion
-        _ <- metastore.update(
-          table,
-          TableChanges(
-            List(
-              UpdatePartitionVersion(Partition(dateCol, "2019-03-01"), version)
+        updateResultAsEither <- metastore
+          .update(
+            table,
+            TableChanges(
+              List(
+                UpdatePartitionVersion(Partition(dateCol, "2019-03-01"), version)
+              )
             )
           )
-        )
+          .redeem(
+            error => Left(error),
+            success => Right(success)
+          )
```

Comment:
There is an `F.fromEither` that does this for you.

Reply:
I think fromEither converts  an `Either[Throwable, A]` into `IO[A]` what I want to do here is kind of the opposite:  convert the `IO[A]` into an `IO[Either[Throwable, A]` so that it will never stop the for comprehension.

Reply:
Ah OK I misread that then. Just out of interest, why do we not want the overall computation to fail in this case?

Reply:
because I added an extra assertion to check the current version hasn't changed. 
So I want the for comprehension to continue so that the current version is retrieved in the next step

Reply:
Ah gotcha, makes sense!